### PR TITLE
[receiver/memcached] Add sample config.yaml and associated test

### LINKED ISSUE #10918: [receiver/memcached] Dead link / missing example config
```
Running target 'checklinks' in module 'receiver/memcachedreceiver' as part of group 'receiver'
/Library/Developer/CommandLineTools/usr/bin/make -C receiver/memcachedreceiver checklinks
command -v markdown-link-check >/dev/null 2>&1 || { echo >&2 "markdown-link-check not installed. Run 'npm install -g markdown-link-check'"; exit 1; }
find . -name \*.md -print0 | xargs -0 -n1 \
                markdown-link-check -q -c /Users/djaglowski/go/github.com/open-telemetry/opentelemetry-collector-contrib/.github/workflows/check_links_config.json || true

FILE: ./README.md
  [✖] ./testdata/config.yaml

  6 links checked.

  ERROR: 1 dead links found!
  [✖] ./testdata/config.yaml → Status: 400
```

### PR DESCRIPTION
**Pull request by djaglowski**
Resolves #10918

### CODE REVIEW

File: receiver/memcachedreceiver/config_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	factories, err := componenttest.NopFactories()
+	require.Nil(t, err)
+
+	factory := NewFactory()
+	factories.Receivers[typeStr] = factory
+	cfg, err := servicetest.LoadConfigAndValidate(filepath.Join("testdata", "config.yaml"), factories)
+	require.NoError(t, err)
+	require.NotNil(t, cfg)
+
+	require.Equal(t, len(cfg.Receivers), 1)
```
Comment (pmm-sumo) [inline receiver/memcachedreceiver/config_test.go:37]: Should we at least check the value associated with endpoint and collection_interval?

**Review by bogdandrutu — APPROVED** [review]

File: receiver/memcachedreceiver/config_test.go (hunk 1 above)
Comment (djaglowski) [reply receiver/memcachedreceiver/config_test.go:44]: We kind of are by comparing the loaded config to the default config. That said, we didn't have any other tests validating that the default config matched a particular expectation, so I added that in a separate test.

File: receiver/memcachedreceiver/config_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/service/servicetest"
+)
+
+func TestDefaultConfig(t *testing.T) {
+	cfg := NewFactory().CreateDefaultConfig().(*Config)
+	require.Equal(t, defaultEndpoint, cfg.Endpoint)
+	require.Equal(t, defaultEndpoint, cfg.Timeout)
+	require.Equal(t, defaultEndpoint, cfg.CollectionInterval)
```
Comment (pmm-sumo) [inline receiver/memcachedreceiver/config_test.go:31]:
```suggestion
	require.Equal(t, defaultCollectionInterval, cfg.CollectionInterval)
```

**Review by pmm-sumo — APPROVED** [review]
Spotted a trivial issue with defaults assigned in test. Allowed myself to fix this quickly to not hold it. :)

**Comment by djaglowski** [comment]
Thanks @pmm-sumo!

**Pull request merged**